### PR TITLE
 Mark all tooltipped text with grey dotted underlines #5696 

### LINKED ISSUE #5696: Mark all tooltipped text with grey dotted underlines
Here is an example:
image

### PR DESCRIPTION
**Pull request by rclakmal**
Fixes #5696

### CODE REVIEW

**Comment by rclakmal** [comment]
Only adds the new css class for span elements. Other wise, buttons,dropdowns also get a dotted line underneath if they have a tooltip. 

Converted all existing abbr tags to span tags so it looks consistent.

File: src/main/webapp/js/common.js [hunk 1]
```diff
@@ -161,6 +161,10 @@ $(document).on('ajaxComplete ready', function() {
     if (isTouchDevice()) {
         $tooltips.tooltip('disable');
     }
+    
+    $('span[data-toggle="tooltip"]').each(function() {
+        $(this).addClass('toolTipDecorate');
+    });
```
Comment (YongJieYongJie) [inline src/main/webapp/js/common.js]: Does it work if you don't use the `.each()` and just use `.addClass` directly?

File: src/main/webapp/stylesheets/teammatesCommon.css [hunk 2]
```diff
@@ -876,3 +876,8 @@ Prevent Long URL From Breaking Out of Container
         white-space: nowrap;
     }
 }
+
+.toolTipDecorate {
```
Comment (YongJieYongJie) [inline src/main/webapp/stylesheets/teammatesCommon.css]: Let's use `.tool-tip-decorate`, and maybe find a better place in the file to put this rule

**Comment by rclakmal** [comment]
@YongJieYongJie Incooperated your comments.

**Comment by YongJieYongJie** [comment]
@rclakmal I think should remove the underline effect for glyphicons like the eye and bell on comments page.

Also, the user email beside the logout button (top-right of page) is similarly underlined because it has a tooltip. Can you check if the tooltip is definitely redundant and can be removed? I think by default the tooltip show just the email, but I'm not sure if there'll ever be useful info.

**Comment by damithc** [comment]
So we show the underline only for tooltipped text? i.e. not for icons or controls.

**Comment by YongJieYongJie** [comment]
@damithc Currently that is the plan, perhaps @rclakmal can try styling the underline for other controls too?

**Comment by damithc** [comment]
Noted @YongJieYongJie Yes, let's start with the text first and see how the UI looks.

**Comment by rclakmal** [comment]
Just to be clear, in this PR we focus on only the textual elements right ? @damithc @YongJieYongJie

**Comment by damithc** [comment]
Yes @rclakmal

**Comment by YongJieYongJie** [comment]
@rclakmal We'll add underline to text elements, so remember to remove the underline from the glyphicons. Also remember to check if the tooltip for the logout button is possibly redundant.

**Comment by rclakmal** [comment]
@YongJieYongJie Now we underline tooltips only if the tooltip value is not the same as what is displayed. This solves both cases you pointed out above (email and icons). Let me know what you think of this approach.

**Comment by YongJieYongJie** [comment]
@rclakmal I think we should actually remove tooltips that are redundant (i.e., tooltip text is substantially the same as displayed text). Were there many redundant tooltips that are also underlined, thus prompting your approach?

**Comment by damithc** [comment]
Yes, redundant ones should be removed. I suspect In some cases, the tool tip can be expanded with more details.

**Comment by rclakmal** [comment]
@YongJieYongJie   @damithc The reason there is a tool-tip for the googleId is, what is displayed is `truncateLongId(googleId)` and the tool-tip has the original googleId. So may be tool-tip could be useful in some cases. 

This is why I went for the above approach where you don't specifically remove or classify(such as icon). If the tool-tip is actually useful (which means it contains something other than displayed), we show the dashed line.

**Comment by YongJieYongJie** [comment]
@rclakmal My concern is that we may be over generalizing for the sake of a few special cases. 

Additionally, this solution is actually not tackling the root of the problem: if the tooltip contain the same content as the text, there shouldn't be a tooltip at all (not just hiding the underline). We should not let users be accustomed to superfluous tooltips.

**Comment by rclakmal** [comment]
> Additionally, this solution is actually not tackling the root of the problem: if the tooltip contain the same content as the text, there shouldn't be a tooltip at all.

@YongJieYongJie Yes, but as I pointed out, in case of a longer email address(googleId), it's getting truncated and only shorter form is displayed. So for some users, the email tooltip could be useful. This could be only decided on run time. To handle this, we could remove the tooltip attribute from the element unless it contains any valuable info.

File: src/main/webapp/js/common.js [hunk 3]
```diff
@@ -161,6 +161,17 @@ $(document).on('ajaxComplete ready', function() {
     if (isTouchDevice()) {
         $tooltips.tooltip('disable');
     }
+    
+    /**
+     * Underlining all span elements with tool-tips except for 
+     * ones without a text value.
+    */
+    $('span[data-toggle="tooltip"]').each(function() {
+        textValue = $(this).text().replace(/\s/g, '')
```
Comment (YongJieYongJie) [inline src/main/webapp/js/common.js]: Perhaps we can add a comment here explaining why we test for empty string. Alternatively, it'll be clearer to explicitly filter out elements that ought to have tooltip but not underline, e.g. `$('span[data-toggle="tooltip"]').not('.glyphicon')...`

File: src/main/webapp/stylesheets/teammatesCommon.css [hunk 4]
```diff
@@ -395,7 +395,15 @@ Apply to thead to make headers not bold.
     border: 0;
 }
 
-/* Code to morph navbar to hamburger at tablet size (992px) instead of phone size (768px) */ 
+/*
+Used for underlining span elements with extra information as titles.
```
Comment (YongJieYongJie) [inline src/main/webapp/stylesheets/teammatesCommon.css]: Let's leave this comment out because the rule is rather simple and the comment actually relates to code elsewhere (and thus risk getting out of sync).

**Comment by rclakmal** [comment]
@YongJieYongJie Incorporated your comments.

**Comment by YongJieYongJie** [comment]
@rclakmal Ready for review?

**Comment by rclakmal** [comment]
@YongJieYongJie Yes :) .

File: src/main/java/teammates/ui/controller/InstructorHomeCourseAjaxPageData.java [hunk 5]
```diff
@@ -142,7 +142,9 @@ private void addAttributeIf(boolean shouldAdd, ElementTag button, String key, St
                     getInstructorHoverMessageForFeedbackSession(session),
                     getInstructorStatusForFeedbackSession(session),
                     TimeHelper.formatDateTimeForInstructorHomePage(session.getStartTime()),
+                    session.getStartTimeString(),
```
Comment (YongJieYongJie) [inline src/main/java/teammates/ui/controller/InstructorHomeCourseAjaxPageData.java]: @rclakmal I think this actually comes from a commit in #5760, let's remove commits for other PRs from this branch so the PRs can be reviewed separately

You may wish to put either PR on hold while finishing the other.

**Comment by rclakmal** [comment]
@YongJieYongJie Sorry about that, it was a mistake done through pulling from the wrong branch. Reverted those changes. Please re-review.

**Comment by YongJieYongJie** [comment]
@rclakmal `src/main/webapp/instructorHelp.html` was removed a while ago and seems to have been accidentially re-added by back to the repository.

File: src/main/webapp/WEB-INF/tags/admin/navBar.tag [hunk 6]
```diff
@@ -60,8 +60,11 @@
                     <a id="btnLogout" class="nav logout" href="<%= Const.ActionURIs.LOGOUT %>">
                         <span class="glyphicon glyphicon-user"></span> Logout
                         
-                        (<span class="text-info" data-toggle="tooltip" data-placement="bottom"
-                                title="${data.account.googleId}">
+                        (<span class="text-info" 
+                                <c:if test="${data.account.googleId != data.account.truncatedGoogleId}">
+                                data-toggle="tooltip" title="${data.account.googleId}"
+                                </c:if>
+                                data-placement="bottom">
```
Comment (YongJieYongJie) [inline src/main/webapp/WEB-INF/tags/admin/navBar.tag]: I think `data-placement` is part of the tooltip and should be in the `<c:if>`

**Comment by rclakmal** [comment]
@YongJieYongJie Thanks for the quick review. Incorporated comments. Also checked for any merge issues via going through change history.

File: src/main/webapp/js/common.js [hunk 7]
```diff
@@ -161,6 +161,18 @@ $(document).on('ajaxComplete ready', function() {
     if (isTouchDevice()) {
         $tooltips.tooltip('disable');
     }
+    
+    /**
+     * Underlining all span elements with tool-tips except for
```
Comment (YongJieYongJie) [inline src/main/webapp/js/common.js]: I think "Underlining" should be reworded as "Underlines"

File: src/main/webapp/stylesheets/teammatesCommon.css [hunk 8]
```diff
@@ -395,7 +395,12 @@ Apply to thead to make headers not bold.
     border: 0;
 }
 
-/* Code to morph navbar to hamburger at tablet size (992px) instead of phone size (768px) */ 
+.tool-tip-decorate {
+	border-bottom: 1px dashed #808080;
+	text-decoration: none;
```
Comment (YongJieYongJie) [inline src/main/webapp/stylesheets/teammatesCommon.css]: Use spaces instead of tab

**Comment by YongJieYongJie** [comment]
@rclakmal Added some comments!

**Comment by rclakmal** [comment]
@YongJieYongJie Fixed.

**Comment by wkurniawan07** [comment]
@rclakmal there is a merge conflict that needs to be resolved here.

**Comment by rclakmal** [comment]
@wkurniawan07 Fixed merge conflicts.
